### PR TITLE
SNAP-1268: Code changes to start SnappyTableStatsProviderService service only once.

### PR DESCRIPTION
**Pull request by snappy-sachin**
## Changes proposed in this pull request

 - SnappyTableStatsProviderService service should be started only once.

## Patch testing

Manually Tested

## ReleaseNotes.txt changes

NA

## Other PRs 

(Does this change require changes in other projects- store, spark, spark-jobserver, aqp? Add the links of PR of the other subprojects that are related to this change)

### CODE REVIEW

**Comment by kneeraj** [comment]
LGTM. A couple of comments.

File: core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala [hunk 1]
```diff
@@ -93,34 +93,38 @@ object SnappyTableStatsProviderService {
...
-              aggregateStats()
-            }
-          } catch {
-            case _: CancelException => // ignore
-            case e: Exception => if (!e.getMessage.contains(
-              "com.gemstone.gemfire.cache.CacheClosedException")) {
-              logger.warning(e)
-            } else {
-              logger.error(e)
+    this.synchronized {
```
Comment (snappy-sachin) [inline core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala]: Done.

File: core/src/main/scala/io/snappydata/SnappyThinConnectorTableStatsProvider.scala [hunk 2]
```diff
@@ -51,24 +51,28 @@ object SnappyThinConnectorTableStatsProvider extends TableStatsProviderService {
...
-    val delay = sc.getConf.getLong(Constant.SPARK_SNAPPY_PREFIX +
-        "calcTableSizeInterval", DEFAULT_CALC_TABLE_SIZE_SERVICE_INTERVAL)
-    doRun = true
-    new Timer("SnappyThinConnectorTableStatsProvider", true).schedule(
-      new TimerTask {
-        override def run(): Unit = {
-          try {
-            if (doRun) {
-              aggregateStats()
+    this.synchronized {
```
Comment (snappy-sachin) [inline core/src/main/scala/io/snappydata/SnappyThinConnectorTableStatsProvider.scala]: Done.

**Comment by snappy-sachin** [comment]
All feedback comments are taken care.
Hence merging..

**Pull request merged**